src/main/java/org/marketplace/services/AdvertisementManagementService.java

```diff
@@ -28,7 +44,11 @@ public Advertisement addAdvertisement(Advertisement advertisement) {
 
     public Advertisement updateAdvertisement(Advertisement advertisement) {
         try {
-            getAdvertisementById(advertisement.getId());
+            Advertisement ad = getAdvertisementById(advertisement.getId());
+            if(ad.getPrice() != advertisement.getPrice()){
+                sendPriceChangeEmail(ad.getUser().getEmail(), ad.getTitle(), advertisement.getPrice());
```

Comment:
This line is a mistake, should sendPriceChangeEmail to all observers, not the owner of the ad.